### PR TITLE
[basicprofiles] Add `precision` parameter to Round Profile

### PR DESCRIPTION
**Pull request by sheilbronn**
## Summary

Add an optional `precision` parameter to the [Round Profile](https://www.openhab.org/addons/transformations/basicprofiles/#round-profile) to support significant-digits rounding for numeric states.

## Changes

- keep existing `scale` parameter behavior
- add optional `precision` parameter
- apply `precision` first and `scale` second when both are configured
- support both `DecimalType` and `QuantityType<?>`
- preserve existing behavior for `UnDefType` and incompatible types
- add config metadata and i18n entries for `precision`

## Details

- `precision` is numeric precision / significant-digits rounding, not display formatting.
- This PR supersedes  #20377, which was aborted due to problems with the commit history/sign-off's.

## Compatibility

This change shall be backward-compatible for existing users of the round profile who only use `scale`.

## Validation

- built the `org.openhab.transform.basicprofiles` bundle successfully
- done: verified formatting with Spotless
- aligned implementation, config class, XML config description, and i18n entries

### CODE REVIEW

File: bundles/org.openhab.transform.basicprofiles/README.md [hunk 1]
```diff
@@ -121,16 +121,20 @@ Switch invertedSwitch { channel="xxx" [profile="basic-profiles:invert"] }
 
 ## Round Profile
 
-The Round Profile scales the state to a specific number of decimal places based on the power of ten.
-Optionally the [Rounding mode](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/math/RoundingMode.html) can be set.
-Source channels should accept Item type `Number`.
+The Round Profile scales the State to a specific number of decimal places based on the power of ten. It can also limit the precision 
+of the State to a specific number of [significant digits](https://en.wikipedia.org/wiki/Significant_figures).
```
Comment (lsiepel) [inline bundles/org.openhab.transform.basicprofiles/README.md]:
```suggestion
The Round Profile scales the State to a specific number of decimal places based on the power of ten.
It can also limit the precision of the State to a specific number of [significant digits](https://en.wikipedia.org/wiki/Significant_figures).
```

File: bundles/org.openhab.transform.basicprofiles/README.md [hunk 2]
```diff
@@ -121,16 +121,20 @@ Switch invertedSwitch { channel="xxx" [profile="basic-profiles:invert"] }
...
+When scaling, a specific [Rounding mode](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/RoundingMode.html) may be applied.
+Source Channels should accept Item Type `Number`.
 
 ### Round Profile Configuration
 
 | Configuration Parameter | Type    | Description                                                                                                     |
 |-------------------------|---------|-----------------------------------------------------------------------------------------------------------------|
-| `scale`                 | integer | Scale to indicate the resulting number of decimal places (min: -16, max: 16, STEP: 1) **mandatory**.            |
-| `mode`                  | text    | Rounding mode to be used (e.g. "UP", "DOWN", "CEILING", "FLOOR", "HALF_UP" or "HALF_DOWN" (default: "HALF_UP"). |
+| `precision`             | integer | Limit the number of significant digits in the output (min: 1, max: 16, STEP: 1).            |
```
Comment (lsiepel) [inline bundles/org.openhab.transform.basicprofiles/README.md]: Please align the table with your favorite markdown tool

Comment (sheilbronn) [reply bundles/org.openhab.transform.basicprofiles/README.md]: Corrected the table formatting.

**Comment by sheilbronn** [comment]
Thanks for reviewing!

File: bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java [hunk 3]
```diff
@@ -59,16 +59,17 @@ public void setup() {
...
+        assertThat(roundProfile.precision, is(4));
         assertThat(roundProfile.scale, is(2));
         assertThat(roundProfile.roundingMode, is(RoundingMode.HALF_UP));
     }
 
     @Test
     public void testDecimalTypeOnCommandFromItem() {
         ProfileCallback callback = mock(ProfileCallback.class);
-        RoundStateProfile roundProfile = createProfile(callback, 2);
+        RoundStateProfile roundProfile = createProfile(callback, 8, 2);
```
Comment (cweitkamp) [inline bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java]: One remark from my side. I would expect that when the new parameter is optional there is no need to touch existing test cases. Otherwise we cannot guarante to not run into an regression for users already using this Profile.

Comment (sheilbronn) [reply bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java]: @cweitkamp I wasn't sure how to deal properly with the situation that `createProfile(callback, <precision>, <scale>, <mode>)` now has a another optional parameter <precision>:
Thats why I chose some value ("8") that shouldn't influence the test result. --> Would "null" be better?

Comment (lsiepel) [reply bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java]: yes, null would be best as that is the case for existing users

**Comment by sheilbronn** [comment]
@cweitkamp @lsiepel on a "sidequest": 
What would you think about **also adding a `div ` and/or `add`parameter** to the Round Profile, allowing the transformation to take care of scaling up/down before the rounding in one step? This could avoid the cumbersome and sometimes impossible chaining of transformations.... 
If this is a worthy discussion I might open a thread and do some first implementation.

File: bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java (hunk 3 above)
Comment (sheilbronn) [reply bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java]: I added another commit to replace `8` and `9` by `null`.

**Comment by lsiepel** [comment]
I have never had the need for a direct math transformation yet, so i can't imagine in what use case one would need this chaining. I guess that in most cases where you need advanced calculations a custom (javascript) transformation script is suitable. I don't think it is worth the effort to add the div or add parameter directly at this moment.

File: bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java (hunk 3 above)
Comment (cweitkamp) [reply bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java]: > yes, null would be best as that is the case for existing users

That is exactly what I meant.

**Comment by sheilbronn** [comment]
@lsiepel 
I used some thing like `div` for transforming incoming percent values and rounding to 1 significant digit in one step, as well as transforming swap sizes (division by 1MiB=1024*1024). But both of these might be considered as rare corner cases....

**Comment by lsiepel** [comment]
They ca be handled by custom transform scripts. Could you fix the build failure?

File: bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java [hunk 4]
```diff
@@ -203,13 +220,16 @@ public void testTimeSeriesFromHandler() {
...
     }
 
-    private RoundStateProfile createProfile(ProfileCallback callback, Integer scale) {
-        return createProfile(callback, scale, null);
+    private RoundStateProfile createProfile(ProfileCallback callback, Integer precision, Integer scale) {
+        return createProfile(callback, precision, scale, null);
     }
 
-    private RoundStateProfile createProfile(ProfileCallback callback, Integer scale, @Nullable String mode) {
+    private RoundStateProfile createProfile(ProfileCallback callback, Integer precision, Integer scale,
```
Comment (cweitkamp) [inline bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java]:
```suggestion
    private RoundStateProfile createProfile(ProfileCallback callback, @Nullable Integer precision, Integer scale,
```

Comment (cweitkamp) [reply bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/RoundStateProfileTest.java]: This will fix the build.
